### DISCUSSION TITLE
Is it working on serverless functions and on serverless edge workers? (Vercel, Cloudflare, Deno Deploy)

### DISCUSSION THREAD

**Discussion by MickL** [adonisjs/core#4471] [Help] [answered]
Hey,
I am wondering if AdonisJS is working on serverless functions and/or edge workers? I didnt find any informations about this.

E.g. I know that Nitro / H3 deploys to "everything", even Deno Deploy but with AdonisJS I am not sure?

Also the UnJS ecosystem (that also powers Nuxt, Nitro, H3) has adapters for nearly everything e.g. the KV abstraction "unstorage" supports 15 drivers including filesystem, Redis, Vercel KV and Cloudflare but I dont see this options here for especially for the serverless options (Vercel, Cloudflare, etc.). E.g. the drive abstraction doesnt have a driver for Vercel Blob and Cloduflare R2 I guess?

**Comment by Barbapapazes** [comment, marked as answer]
> Is it working on serverless functions and on serverless edge workers?

short answer, no. long answer: Adonis is not designed to be deployed on Edge because it's not the purpose of a framework like this and the overhead of serverless platform is way too heavy for the gain.

For the driver, the version for v6 is not yet release but in v5, writing an adapter was not complicated and FS and S3 was available.

**Reply by MickL** [reply]
Thanks for the answer!

Btw. "A framework like this" is also [Nitro](https://nitro.unjs.io/) and it deploys everywhere, even on edge :)

**Reply by RomainLanz** [reply]
lol.

Nitro is nothing like AdonisJS.
Please read [this](https://github.com/adonisjs/core/discussions/1256#discussioncomment-38594). The answer hasn't changed.

**Reply by Barbapapazes** [reply]
Team member of UnJS here 👋

You're comparing a F1 vs an off-road car asking if the F1 car could become off road. Maybe with time, it could be possible but you'll loose every benefits of the F1 car and while you'll be able to have a F1 off road, you'll not develop the F1.

Choosing between Adonis and Nitro is a thing and it really depends of your needs but it's impossible to answer to "which one is the best". However, and for all the love I have for Nitro, it's actually impossible to achieve the same things and they are not comparable. For an enterprise grade application, Adonis is the best for a lot of reasons like the fully featured part (auth, cache, view, migrations...).

Nitro is designed for cold start (under 2ms) and to be a toolkit for other framework but under the hood, it's very complicated, from maintaining presets to building runtime agnostic packages like db0 or crossws. 

Finally, if you think that deployment is THE thing that make a framework, you're wrong.

**Reply by MickL** [reply]
I wasnt comparing, even tho I think they are very comparable. Sure, AdonisJS has the batteries included but thats about it. The things you said are available for Nitro and all of them run serverless too, e.g. auth with Lucia, Cache is integrate in Nitro, Views and Templates are not what I expect from a backend but what is better than Nuxt powered by Nitro? and migrations are part of Drizzle. Nothing is missing for Nitro, everything runs blazing fast serverless and actually Nitro / UnJS has way more drivers for was more platforms and providers. And yes for me deployment is an important factor. But this wasnt supposed to be a comparison.

In any case my question was just if AdonisJS is deployable serverless and the answer is no. Thanks.
